sandbox/src/schema/typeGraphqlSchema.ts

```diff
@@ -0,0 +1,1042 @@
+  name!: string;
+
+  @Field((type) => String, {})
+  rulesetText!: string;
+}
+
+@InputType({
+  description: `PreCoordinatedConstraintTypeInput`,
+})
+export class PreCoordinatedConstraintTypeInput {
+  @Field((type) => String, { nullable: true })
+  id!: string;
+
+  @Field((type) => String, { nullable: true })
+  name!: string;
+
+  @Field((type) => String, { nullable: true })
+  rulesetText!: string;
+}
+
+@ObjectType({
+  description:
+    'Some conditions do not fit well in the model and are handled as special cases. This included Morbid Obesity. The terms pre and post coordinated can be read about here https://blog.clinicalarchitecture.com/informatics-lingo-pre-and-post-coordinated-terms',
+})
+export class PreCoordinatedConstraint {
+  @Field((type) => String, {})
+  id!: string;
+
+  @Field((type) => PreCoordinatedConstraintType, {
+    description: 'PreCoordinatedConstraintType',
+  })
+  preCoordinatedConstraintType!: PreCoordinatedConstraintType;
+}
+
+@InputType({
+  description:
+    'Some conditions do not fit well in the model and are handled as special cases. This included Morbid Obesity. The terms pre and post coordinated can be read about here https://blog.clinicalarchitecture.com/informatics-lingo-pre-and-post-coordinated-terms',
+})
+export class PreCoordinatedConstraintInput {
+  @Field((type) => PreCoordinatedConstraintTypeInput, {
+    description: 'PreCoordinatedConstraintType',
+  })
+  preCoordinatedConstraintType!: PreCoordinatedConstraintTypeInput;
+}
+
+@ObjectType({
+  description:
+    "A Rule composes a collection of Code Lists with a number of 'constraints' (e.g. people over 5 years of age) which act together to control which people are captured.",
+})
+export class Rule {
+  @Field((type) => String, {})
+  id!: string;
+
+  @Field((type) => [CodeList], {
+    description:
+      'A Code List aggregates individual Codes into logical groupings of codes. These are used to identify patients to be included in a Cohort (and in some instances people to be excluded)',
+    nullable: true,
+  })
+  codeListInclusions?: CodeList[];
+
+  @Field((type) => [CodeList], {
+    description:
+      'A Code List aggregates individual Codes into logical groupings of codes. These are used to identify patients to be included in a Cohort (and in some instances people to be excluded)',
+    nullable: true,
+  })
+  codeListsResolving?: CodeList[];
+
+  @Field((type) => [PatientAgeConstraint], {
+    description:
+      'This constrains the people identified to be within certain age limits. This might be used to find people over 5 years of age.',
+    nullable: true,
+  })
+  patientAgeConstraints?: PatientAgeConstraint[];
+
+  @Field((type) => EventDateConstraint, {
+    description:
+      'This constrains the date when the patient journal event occurred. This might be to identify people who had a certain treatment in the last two years.',
+    nullable: true,
+  })
+  eventDateConstraint?: EventDateConstraint|null;
```

Comment:
Can you please add a comment explaining why this differs from the original schema?

Reply:
Done